### ISSUE TITLE
split_by_extenstion might return wrong extenstion

### ISSUE THREAD

**Issue by fcharlie** [gabime/spdlog#589]
https://github.com/gabime/spdlog/blob/f5939f9e56ceda7f6d071da8498a1c973ed8eac5/include/spdlog/details/file_helper.h#L122-L130

When input a path like:
```
rc.d/service
```
split_by_extenstion will return extenstion `d/serivce`. so must rfind '/' first.

**Comment by fcharlie** [comment]
```cpp
static std::tuple<filename_t, filename_t>
split_by_extenstion(const filename_t &fname) {
  auto index = fname.rfind('.');
  if (index != filename_t::npos && index != fname.size() - 1) {
    auto index2 = fname.find('/', index);
    if (index2 == fname.npos) {
      return std::make_tuple(fname.substr(0, index), fname.substr(index));
    }
  }
  return std::make_tuple(fname, std::string());
}
```

**Comment by godbyk** [comment]
I would probably do this farther up the call stack. That is, document that `split_by_extenstion()` takes a filename only (not a file path). And the caller is responsible for calling `basename()` or equivalent and only pass in the filename component.

**Comment by fcharlie** [comment]
@godbyk Mybe change this comment and args
https://github.com/gabime/spdlog/blob/f5939f9e56ceda7f6d071da8498a1c973ed8eac5/include/spdlog/details/file_helper.h#L111-L120

**Comment by godbyk** [comment]
@fcharlie Good point! In that case, your solution is better. A few other thoughts:

 * You're missing an `&&` operator in the first `if` statement.
 * I would suggest using `details::os::folder_sep` instead of `'/'` so that it works in Windows, too.
 * Can we fix the typo in the function name while we're at it?
 * Could you add your new test cases to the `tests/file_helper.cpp` unit tests?

**Comment by fcharlie** [comment]
@godbyk I update it.

**Comment by gabime** [comment]
@fcharlie Good catch! I think I need to rearrange the code a little in file_helper.h - by adding a new function called “basename(..)” and use it in the “split_by_extension” implementation.

**Comment by fcharlie** [comment]
Yes, But I think `'/'` under Windows is not invalid
```
C:/tmp/spdlog.log
```
And:
```
C:/tmp//need//spdlog.log
```

**Comment by gabime** [comment]
Fixed in 0.16.2
